abrams_strogatz/lang_comp_bilinguals.py

```diff
@@ -0,0 +1,300 @@
+    the program will check if the selected node must change his/her
+    language. After a number of iterations (selected by the developer),
+    this function saves a snapshot of the language dynamics.
+
+    Args:
+        popu: Array containing the language spoken by the population.
+            The values contained inside the array are 1 (lang A), 
+            0 (bilingual) and -1 (lang B).
+        ph: Integer number labelling the representation. Describes the
+            number of finished iterations (iterations = ph * mult).
+        key: Boolean value. If the key = True, the program calls the
+            function. Defaults to True
+
+    Returns:
+        The function saves a png file inside the directory where the
+        program is executed.
+
+    """
+    if key:
+        figr = repres(popu)
+        figr.savefig("lang_comp_{}.png".format(str(ph)), bbox_inches="tight")
+
+
+# Parameters
+m = 20
+n = 20
+a = 1.0
+s = 0.5
+pa = 1./3
+pb = 1./3
+# Initial condition
+popu = initial_cond(m, n, pa, pb)
+initial_popu = popu.copy()
+# Check error
+na = np.count_nonzero(popu[popu > 0])
+nb = np.count_nonzero(popu[popu < 0])
+nab = (popu == 0).sum()
+print("first error") if nb != (m * n - na - nab) else None
+print("num_A = {0}, num_B = {1}, num_AB = {2}".format(na, nb, nab)) 
+# Iterative process
+mult = 15
+maxit = mult * 400  # Maximum number of iterations. Information is stored
+# every mult steps
+num_speakers = np.ones(shape=(int(maxit / mult), 3))  # Container
+
+for iterr in np.arange(maxit):
+    popu = language_dynamics(popu, m, n, s)
+    # Saving process and number of speakers
+    if (iterr % mult) == 0:
+        ph = int(iterr / mult)
+        numa = np.count_nonzero(popu[popu > 0])
+        numb = np.count_nonzero(popu[popu < 0])
+        numab = (popu == 0).sum()
+        print("error") if numb != (m * n - numa - numab) else None
+        num_speakers[ph] = np.array([numa, numb, numab])
+        saving_process(popu, ph, key=False)
+    else:
+        None
+
+# Representation
+color = mpl.colors.ListedColormap(["Blue", "White", "Red"])
+colbar_tick = np.array([-1, 0, 1])
+ratio = popu.shape[0] / popu.shape[1]
+# Figures
+fig, axs = plt.subplots(nrows=2, ncols=2)
+gs = axs[1, 0].get_gridspec()
+fig.suptitle("Language competition")
+# First and second plots
+for col in range(2):
+    ax = axs[0, col]
+    grid = initial_popu if (col == 0) else popu
+    title = "Initial state" if (col == 0) else "Final state"
+    plot = ax.matshow(grid, cmap=color, origin="lower", animated=True)
+    ax.xaxis.tick_bottom()
+    ax.xaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.yaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.xaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.yaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.set_title(title)
+    fig.colorbar(
+        plot, ax=ax, ticks=colbar_tick, label="Language", fraction=0.047 * ratio
```

Comment:
261:  81 C815 missing trailing comma in Python 3.5+ [flake8-commas]
```suggestion
        plot, ax=ax, ticks=colbar_tick, label="Language", fraction=0.047 * ratio,
```